### ISSUE TITLE
SelectOnFocus for text fields (TextArea, TextField) is ignored (always true)

### ISSUE THREAD

**Issue by fabriciomurta** [extnet/Ext.NET#1441]
Found: 4.2.0
Ext.NET Thread: [SelectText problem](http://forums.ext.net/showthread.php?61783)
Sencha thread (assumed as fixed but seems fix is not perfect): [ExtJS 6.0.2: Text fields focus on select regardless of selectOnFocus config](https://www.sencha.com/forum/showthread.php?310439)

In either `Ext.Net.TextArea` or `Ext.Net.TextField`, focusing the field always selects all the text in the input field, ignoring the `SelectOnFocus` setting. The sencha thread above says it has been fixed in ExtJS as of 6.0.3 (thus 6.2.0 and 6.2.1) but according to [post #8 on sencha thread above](https://www.sencha.com/forum/showthread.php?310439&p=1170408&viewfull=1#post1170408), the fix also introduces an issue where the cursor is always moved to the end -- which would likely prevent the `selectText()` method from working.

The same post mentioned above also suggested an override that works nicely with the test case provided. A reviewed test case with the provided in Ext.NET thread and the override from Sencha thread would be:

```xml
<%@ Page Language="C#" %>

<!DOCTYPE html>

<html>
<head runat="server">
    <title>61783v4 - selectText() issue</title>
    <script type="text/javascript">
        // solution from: https://www.sencha.com/forum/showthread.php?310439&p=1170408&viewfull=1#post1170408
        Ext.define('MyApp.overrides.form.field.Text', {
            override: 'Ext.form.field.Text',
            onFocus: function (e) {
                var me = this,
                    inputEl = me.inputEl.dom,
                    startValue, value, len;

                // Changing call to callSuper instead of callParent to skip the original method
                me.callSuper([e]);

                // Added check for selectOnFocus
                if (me.selectOnFocus) {
                    // This handler may be called when the focus has already shifted to another element;
                    // calling inputEl.select() will forcibly focus again it which in turn might set up
                    // a nasty circular race condition if focusEl !== inputEl.
                    if (!me.focusTimer) {
                        startValue = inputEl.value;
                        me.focusTimer = Ext.asap(function () {
                            me.focusTimer = null;
                            // This ensures the carret will be at the end of the input element
                            // while tabbing between editors.
                            if (!me.destroyed && document.activeElement === inputEl) {
                                value = inputEl.value;
                                len = value.length;

                                // If focusing has fired an event which mutated the value,
                                // place the caret at the end. Else select the initial text
                                // as is the HTML default behaviour.
                                me.selectText(value !== startValue ? len : 0, len);
                            }
                        });
                    }
                }


                if (me.emptyText) {
                    me.autoSize();
                }

                me.addCls(me.fieldFocusCls);
                me.triggerWrap.addCls(me.triggerWrapFocusCls);
                me.inputWrap.addCls(me.inputWrapFocusCls);
                me.invokeTriggers('onFieldFocus', [e]);
            }
        });
    </script>
</head>
<body>
    <ext:ResourceManager runat="server"/>
    <ext:TextArea
        ID="txtA1"
        runat="server"
        Height="70"
        Width="710"
        MaxLength="1000"
        EnforceMaxLength="true"
        Text="lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum "
        SelectOnFocus="true"
         />

    <ext:Button ID="Button2" runat="server" Text="Select" >
        <Listeners>
            <Click Handler="App.txtA1.focus(); App.txtA1.selectText(5,9);" />
            <Focus Handler="return false;" />
        </Listeners>
    </ext:Button>
</body>
</html>
```

The Sencha forum thread evolves for having the issue partly fixed in a nightly build after 6.2.1 by jan 10th 2017, and support team double checks the bug has been fixed in the nightly by feb 9th 2017.

**Comment by fabriciomurta** [comment]
The fix above has been applied on Ext.NET.

**Comment by fabriciomurta** [comment]
Well leave it open so that we remember to review on next ExtJS merge.

**Comment by fabriciomurta** [comment]
**Update:** Sencha issue _EXTJS-20985_ allegedly fixed in ExtJS 6.0.3 release.

**Comment by fabriciomurta** [comment]
**Update:** Sencha issue _EXTJS-20985_ allegedly fixed in ExtJS 6.0.3 release.

**Comment by pescobargisystems** [comment]
Thanks for the solution! It has worked for me. Greetings.
